Dockerfile.debian-slim

```diff
@@ -0,0 +1,71 @@
+FROM debian:10-slim AS build
+
+RUN apt-get update \
+    && apt-get install -y libssl-dev \
+    llvm-dev clang libclang-dev \
+    wget git cmake build-essential pkg-config
+
+# Rust setup
+ARG RUST_VERSION=1.43.0
+
+ENV RUSTUP_HOME=/usr/local/rustup \
+    CARGO_HOME=/usr/local/cargo \
+    PATH=/usr/local/cargo/bin:$PATH
+
+RUN set -eux; \
+    dpkgArch="$(dpkg --print-architecture)"; \
+    case "${dpkgArch##*-}" in \
+        amd64) rustArch='x86_64-unknown-linux-gnu'; rustupSha256='ad1f8b5199b3b9e231472ed7aa08d2e5d1d539198a15c5b1e53c746aad81d27b' ;; \
+        armhf) rustArch='armv7-unknown-linux-gnueabihf'; rustupSha256='6c6c3789dabf12171c7f500e06d21d8004b5318a5083df8b0b02c0e5ef1d017b' ;; \
+        arm64) rustArch='aarch64-unknown-linux-gnu'; rustupSha256='26942c80234bac34b3c1352abbd9187d3e23b43dae3cf56a9f9c1ea8ee53076d' ;; \
+        i386) rustArch='i686-unknown-linux-gnu'; rustupSha256='27ae12bc294a34e566579deba3e066245d09b8871dc021ef45fc715dced05297' ;; \
+        *) echo >&2 "unsupported architecture: ${dpkgArch}"; exit 1 ;; \
+    esac; \
+    url="https://static.rust-lang.org/rustup/archive/1.21.1/${rustArch}/rustup-init"; \
+    wget "$url"; \
+    echo "${rustupSha256} *rustup-init" | sha256sum -c -; \
+    chmod +x rustup-init; \
+    ./rustup-init -y --no-modify-path --profile minimal --default-toolchain $RUST_VERSION; \
+    rm rustup-init; \
+    chmod -R a+w $RUSTUP_HOME $CARGO_HOME; \
+    rustup --version; \
+    cargo --version; \
+    rustc --version;
+
+# Ruby setup
+ARG RUBY_VERSION=2.6.3
+ARG RUBY_INSTALL_VERSION=0.7.0
+ARG RUBY_INSTALL_SHA=500a8ac84b8f65455958a02bcefd1ed4bfcaeaa2bb97b8f31e61ded5cd0fd70b
+
+RUN set -eux; \
+    echo 'gem: --no-document' >> /usr/local/etc/gemrc; \
+    url="https://github.com/postmodern/ruby-install/archive/v$RUBY_INSTALL_VERSION.tar.gz"; \
+    wget -O /tmp/ruby-install.tar.gz "$url"; \
+    echo "${RUBY_INSTALL_SHA} /tmp/ruby-install.tar.gz" | sha256sum -c -; \
+    tar -xzvf /tmp/ruby-install.tar.gz -C /opt && rm /tmp/ruby-install.tar.gz; \
+    /opt/ruby-install-$RUBY_INSTALL_VERSION/bin/ruby-install -i /usr/local/ ruby $RUBY_VERSION -- --disable-install-rdoc; \
+    rm /usr/local/src/ruby-$RUBY_VERSION.tar.bz2; \
+    gem update --system; \
+    ruby --version; \
+    gem --version; \
+    bundle --version;
+
+# XXX HACK - the following build arg is used primarily to break caching and force rebuild, since
+# we're bulding nightlies here and fetching something like HEAD commit SHA will complicate things
+ARG ARTICHOKE_NIGHTLY_VER=latest
+# Install artichoke
+RUN set -eux; \
+    cargo install --git https://github.com/artichoke/artichoke --locked artichoke; \
+    airb --version; \
+    artichoke --version;
+
+# Setup runtime
+FROM debian:10-slim AS runtime
```

Comment:
Can we name the image `debian:buster-slim`. I find Debian releases easier to identify by their code name.

This is not necessary for Ubuntu because the release numbers are meaningful.